packages/ui/src/components/Tabs/Tabs.tsx

```diff
@@ -48,11 +49,14 @@ function Tabs({
       : ''
   )
 
+  const router = useRouter()
+  const hash = router.asPath.split('#')[1]?.toUpperCase()
+
   let __styles = styleHandler('tabs')
 
   // activeId state can be overriden externally with `active`
-  // defaults to use activeTab
-  const active = activeId ? activeId : activeTab
+  // defaults to use a url #hash if we have one or activeTab if not
+  const active = hash ? hash : activeId ? activeId : activeTab
```

Comment:
nit: i think `activeId` should take precedence over `hash` and `activeTab`

if we're passing `activeId` as a prop into `<Tabs>` -> it's expected that the corresponding tab should be highlighted no matter the hash